Zapper-fi/studio
# feat(opensky-finance): Add OpenSky Finance integration

## Description
Add OpenSky Finance integration. (OpenSky Finance is a time-based NFT lending protocol built on top of aave.com which integrates peer-to-pool and peer-to-peer loans.)

* Supply position fetcher (AppToken type)
* Borrow position fetcher (loan is represented not ERC20 but NFT in OpenSky protocol, so we select 'ContractPosition' type)
* Supply balance fetcher

## Checklist

- [x] (optional) As a contributor, my Twitter handle is:  [OpenSkyFinance](https://twitter.com/OpenSkyFinance)

## How to test?


1. Edit/create .env adding `ENABLED_APPS=opensky`
2. Run `pnpm dev` to spin up nest server
3. Visit dev endpoints below to see result of fetchers to integrate ferro:

* http://localhost:5001/apps/opensky/tokens?groupIds[]=supply&network=ethereum
* http://localhost:5001/apps/opensky/positions?groupIds[]=borrow&network=ethereum
* `http://localhost:5001/apps/opensky/balances?addresses[]=<wallet_address>&network=ethereum`
